src/overlays/actors/ovl_En_Pm/z_en_pm.c

```diff
@@ -1132,11 +1132,11 @@ void func_80AF8BA8(s32 arg0) {
 
 void func_80AF8C68(EnPm* this, PlayState* play) {
     Player* player = GET_PLAYER(play);
-    s32 sp28 = Message_GetState(&play->msgCtx);
+    s32 talkState = Message_GetState(&play->msgCtx);
     u16 temp_a0 = play->msgCtx.currentTextId;
 
-    if ((player->targetActor == &this->actor) && ((temp_a0 < 255) || (temp_a0 > 512)) && (sp28 == 3) &&
-        (this->unk_388 == 3)) {
+    if ((player->targetActor == &this->actor) && ((temp_a0 < 255) || (temp_a0 > 512)) && (talkState == TEXT_STATE_3) &&
```

Comment:
```suggestion
    if ((player->targetActor == &this->actor) && ((temp_a0 < 0xFF) || (temp_a0 > 0x200)) && (talkState == TEXT_STATE_3) &&
```
It's that weird condition again.